### PR TITLE
Update license

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

Added Python.NET, removed unused libraries, and updated some links.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @QilongTang

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -41,13 +44,13 @@ https://www.gnu.org/software/gettext/
 https://www.gnu.org/software/gettext/manual/html_node/GNU-LGPL.html#GNU-LGPL
 
 ### Helix Toolkit
-https://helixtoolkit.codeplex.com/
-https://helixtoolkit.codeplex.com/license 
+https://github.com/helix-toolkit/helix-toolkit
```
Comment (QilongTang) [inline LICENSE.txt:47]: Wonderful

File: doc/distrib/License.rtf [hunk 2]
```diff
@@ -1,526 +1,119 @@
...
+\par
+\cf5\b\f0 StarMath\par
+{\cf0\b0\f2{\field{\*\fldinst{HYPERLINK https://github.com/DesignEngrLab/StarMath/blob/master/LICENSE }}{\fldrslt{https://github.com/DesignEngrLab/StarMath/blob/master/LICENSE\ul0\cf0}}}}\cf0\b0\f2\fs22\par
+\f3\par
+\cf1\b\f0 Xceed Extended WPF Toolkit\f1\par
+{{\field{\*\fldinst{HYPERLINK "https://opensource.org/licenses/ms-pl.html"}}{\fldrslt{\ul\cf4\cf0\b0\f2 Microsoft Public License}}}}\cf0\b0\f3\fs22\par
+{\f2{\field{\*\fldinst{HYPERLINK https://github.com/xceedsoftware/wpftoolkit/blob/master/license.md }}{\fldrslt{https://github.com/xceedsoftware/wpftoolkit/blob/master/license.md\ul0\cf0}}}}\f3\fs22\par
+
+\pard\nowidctlpar\sb240\par
+}
```
Comment (QilongTang) [inline doc/distrib/License.rtf:118]: For some reason seems there are way more content in the previous version? Is this IDE related? @mjkkirschner  should know more here

Comment (mjkkirschner) [reply doc/distrib/License.rtf:118]: yeah, the same RTF view can be created with different elements of course - we should really test this in the application though as .net renders rtf kinda funny.

Comment (mmisol) [reply doc/distrib/License.rtf:118]: I noticed that too. I initially edited the file in WordPad and opened it in Dynamo -> About. Things looked good except for a few links I added that were invisible! To fix that I had to play with the "RTF tags" until they decided to appear.

File: LICENSE.txt [hunk 3]
```diff
@@ -12,16 +14,13 @@ Unless required by applicable law or agreed to in writing, software distributed
 ## Third Party Licenses ##
 
 ### Avalon Edit
-http://www.codeproject.com/Articles/42490/Using-AvalonEdit-WPF-Text-Editor
-http://opensource.org/licenses/lgpl-3.0.html
+http://avalonedit.net/
+https://opensource.org/licenses/lgpl-2.1.php
+  (LGPL-2.1 license)
```
Comment (mjkkirschner) [inline LICENSE.txt:19]: it looks like AvalonEdit is actually MIT:
https://github.com/icsharpcode/AvalonEdit/blob/master/LICENSE

File: LICENSE.txt [hunk 4]
```diff
@@ -123,25 +110,28 @@ http://msdn.microsoft.com/en-us/library/gg405489(PandP.40).aspx
 
 ### Python Standard Library
 https://www.python.org/
-https://www.python.org/download/releases/2.7/license/
```
Comment (mjkkirschner) [inline LICENSE.txt:126]: we are still shipping the 2.7 std lib and so I think this should remain.

File: LICENSE.txt (hunk 3 above)
Comment (mmisol) [reply LICENSE.txt:19]: Lates version is, but the one we are using is LGPL-2.1. See https://www.nuget.org/packages/AvalonEdit/4.3.1.9430

File: LICENSE.txt (hunk 4 above)
Comment (mmisol) [reply LICENSE.txt:126]: Actually, the link I replaced this with includes the license of every Python version up to this day. It's an updated version of the old link, which included the license of every Python version up to 2.7.

File: LICENSE.txt (hunk 3 above)
Comment (mjkkirschner) [reply LICENSE.txt:19]: hmm, I wonder if it is worth it to check which version Revit ships these days and align if they have moved up to a more recent one.

Comment (mmisol) [reply LICENSE.txt:19]: That's a good question. I guess currently we rely on them to test, but if it works we don't know if we could have upgraded. In that case we would be sticking with an old version for no reason.

Can we maybe run the D4R tests and see if causes problems? Or is it more complex than that?

**Comment by mjkkirschner** [comment]
Let me just dig up their manifest for the version they use - testing d4r unfortunately won’t catch the conflict
